### PR TITLE
reference/configuration: add docs for spilling

### PR DESCRIPTION
**Pull request by SunRunAway**
### What is the related PR or file link(s)? 

- Reference link(s): https://github.com/pingcap/tidb/pull/14600
- This PR is to align with:https://github.com/pingcap/docs/pull/1871
- [ ] N/A (not applicable)

**Important Notice:**

**If your changes apply to multiple TiDB documentation versions**, to trigger the bot to cherry-pick this PR to other release branches, make sure you add labels such as:

- **needs-cherry-pick-3.0**
- **needs-cherry-pick-3.1**
- **needs-cherry-pick-2.1**

### CODE REVIEW

File: reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -15,18 +15,29 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
+### `tmp-storage-path`
+
++ 超出 `mem-quota-query` 时，某些算子会使用的临时磁盘存储位置。
++ 默认值：`<操作系统临时文件夹>/tidb/tmp-storage`
+
+### `oom-action`
+
++ 当 TiDB 中的某条 SQL 超出 `mem-quota-query` 并且也不能再利用临时磁盘时的行为。
++ 默认值："log"
++ 现在合法的选项是 ["log", "cancel"]，如果为 "log"，仅仅是打印日志，不作实质处理。如果为 "cancel"，我们会取消执行这个 SQL 操作，并且输出日志。
```
Comment (anotherrachel) [inline reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 当 TiDB 中单条 SQL 语句超出内存限制且不能启用临时磁盘时的行为。
+ 默认值："log"
+ 目前合法的选项为 ["log", "cancel"]。设置为 "log" 时，仅输出日志。设置为 "cancel" 时，取消执行该 SQL 操作，并输出日志。
```

File: reference/configuration/tidb-server/configuration.md [hunk 2]
```diff
@@ -80,6 +80,11 @@ category: reference
 + 默认："/tmp/tidb"
 + 可以通过 `tidb-server --store=mocktikv --path=""` 来启动一个纯内存引擎的 TiDB
 
+## `--tmp-storage-path`
+
++ TiDB 临时磁盘存储位置。
++ 默认：`<操作系统临时文件夹>/tidb/tmp-storage`
```
Comment (zz-jason) [inline reference/configuration/tidb-server/configuration.md]: we should also mention that this config item takes effect only when `oom-use-tmp-storage` is set to `true`

Comment (SunRunAway) [reply reference/configuration/tidb-server/configuration.md]: It's already stayed here. See https://github.com/pingcap/docs-cn/pull/2306/files#diff-03d37f540e23cadd32b8e1c4c20d4b76R33

Comment (SunRunAway) [reply reference/configuration/tidb-server/configuration.md]: I'll remove this config in another PR.

File: reference/configuration/tidb-server/configuration-file.md [hunk 3]
```diff
@@ -15,18 +15,29 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 ### `mem-quota-query`
 
 + 单条 SQL 语句可以占用的最大内存阈值。
 + 默认值：34359738368
 + 超过该值的请求会被 `oom-action` 定义的行为所处理。
 
+### `oom-use-tmp-storage`
+
++ 设置是否在单条 SQL 语句的内存使用超出 `mem-quota-query` 限制时为某些算子启用临时磁盘。
++ 默认值: true
```
Comment (anotherrachel) [inline reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认值：true
```

**Pull request merged**